flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/TimestampsAndWatermarksOperator.java

```diff
@@ -64,24 +65,30 @@
 	/** The interval (in milliseconds) for periodic watermark probes. Initialized during runtime. */
 	private transient long watermarkInterval;
 
-	public TimestampsAndWatermarksOperator(
-			WatermarkStrategy<T> watermarkStrategy) {
+	/** Whether to periodically emit watermarks or only one final watermark at the end of input.*/
```

Comment:
It is not only about periodic. I would say `wheter to emit intermediate watermarks or only one ...`